src/MSBuild/TerminalLogger/SystemStopwatch.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace Microsoft.Build.Logging.TerminalLogger;
+
+internal sealed class SystemStopwatch : StopwatchAbstraction
```

Comment:
Not your problem, but thinking about abstractions like this that are useful for testing but mean that in release code, you have an object that points to another object and doesn't really have any further purpose, I'm wondering if it would be a nice (runtime?) feature to essentially inline an object if it doesn't do anything but send redirect requests to one specific inner object.

Reply:
The JIT can indeed inline method calls in many cases, and (without actually checking) I'd expect that here, especially post Dynamic PGO which should see that at (real non-test) runtime there's only ever one type of `StopwatchAbstraction` used.